Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,433 @@
+               LetRefNode=object, TreeFragment=object, EncodedString=object,
+               error=object, warning=object, copy=object, _unicode=object)
+
+
+# Inspired by and based around https://github.com/cython/cython/pull/3818
+# with some less lazy changes to it and a few minor improvements and optimizations...
+
+# Decided to revert to an older variant I had wrote of this code for the sake of 
+# maintainability - Vizonex
+
+
+
+# TODO Save this implementation commented out if required....
+if sys.version_info >= (3, 9):
+    typing_module = "typing"
+else:
+    typing_module = "typing_extensions"
+
+class PyiWriter(CythonTransform, DeclarationWriter):
+    """Used By Cython to help Write stubfiles
+    this comes in handy for ides like Pylance 
+    which suffer from having no code access to 
+    annotations from compiled python modules...
+    """
+
+    def __init__(self, context):
+        super(PyiWriter, self).__init__(context=context)
+        super(DeclarationWriter, self).__init__()
+        self.context = context
+        self.module_name = ""
+        self.class_func_count = 0 
+
+        self.translation_table = {}
+        """Used as an eternal resource for translating ctype declarations into python-types"""
+
+        self.use_typing = False
+        """if true we must import typing's generator typehint..."""
+
+
+    def _visitchildren_indented(self, node):
+        self.indent()
+        self.visitchildren(node)
+        self.dedent()
+    
+    def translate_pyrex_type(self, ctype):
+        # TODO implement Pyrex to cython shadow typehints converter...
+
+        if isinstance(ctype, PyrexTypes.BuiltinObjectType):
+            return ctype.py_type_name()
+
+        if isinstance(ctype, PyrexTypes.CVoidType):
+            if ctype.is_ptr:
+                return "object"
+            return "None"
+        
+        if isinstance(ctype, PyrexTypes.CIntType):
+            return "int"
+
+        elif isinstance(ctype, PyrexTypes.CFloatType):
+            return "float"
+
+        elif isinstance(ctype,PyrexTypes.PyObjectType):
+            return ctype.py_type_name()
+        
+        return 'object'
+
+
+    # Instead of doing it into C, we're doing it backwards...
+    def translate_base_type_to_py(
+        self,
+        base
+        ):
+
+        # Try checking our table first...
+        if self.translation_table.get(base.name):
+            return self.translation_table[base.name]
+
+        elif base.name == "object":
+            return "object"
+
+        elif base.name in ("unicode","basestring"):
+            return "str"
+
+        elif not base.is_basic_c_type:
+            # Likely that it's already a python object that's being handled...
+            # except for basestring and unicode...
+            return base.name 
+
+        elif base.name == "bint":
+            return "bool"
+
+        ctype = PyrexTypes.simple_c_type(base.signed, base.longness, base.name) # type: ignore
```

Comment:
Perhaps I'm missing something, but `base` is already a type, so I don't understand what this line does